engine/collection/compliance/core.go

```diff
@@ -301,7 +301,7 @@ func (c *Core) processBlockProposal(proposal *messages.ClusterBlockProposal) err
 
 	// submit the model to hotstuff for processing
 	log.Info().Msg("forwarding block proposal to hotstuff")
-	c.hotstuff.SubmitProposal(header, parent.View)
+	<-c.hotstuff.SubmitProposal(header, parent.View)
```

Comment:
We always wait here. There doesn't seem to be an easy way to check if the block is part of a range response. Maybe this isn't an issue but I just want to confirm.

Previous PR
- Identify range response: https://github.com/onflow/flow-go/pull/2721/files#diff-08f436e5039dd8c83b975482d9c1126337bf36bce71b5535116129eb580ad8c4R146
- Conditional wait: https://github.com/onflow/flow-go/pull/2721/files#diff-08f436e5039dd8c83b975482d9c1126337bf36bce71b5535116129eb580ad8c4R393

Reply:
Please see my comment here
https://github.com/onflow/flow-go/pull/2721/files#r929442712